src/api/services/ListingItemTemplateService.ts

```diff
@@ -419,6 +426,40 @@ export class ListingItemTemplateService {
         return messageSize;
     }
 
+    // sets an image as a "featured" image
+    public async setFeaturedImg(listingItemTemplate: resources.ListingItemTemplate, imageID: number): Promise<void> {
+        try {
+            const templateID = listingItemTemplate.id;
+            const itemImages = listingItemTemplate.ItemInformation.ItemImages;
+            if (!itemImages.find((img) => img.id === imageID)) {
+                throw new MessageException('Image ID doesnt exist on template');
+            }
+            if (itemImages) {
+                // loop through templates to check for previous featured images, sets to false
+                for (const item of itemImages) {
```

Comment:
instead of looping through all, you could use array.find as you previously did..